src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.NativeAot.Windows.cs

```diff
@@ -0,0 +1,103 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Text;
+
+namespace System.Runtime.InteropServices
+{
+    public static partial class NativeLibrary
+    {
+        private static IntPtr LoadLibraryHelper(string libraryName, int flags, ref LoadLibErrorTracker errorTracker)
+        {
+            IntPtr ret = Interop.Kernel32.LoadLibraryEx(libraryName, IntPtr.Zero, flags);
+            if (ret != IntPtr.Zero)
+            {
+                return ret;
+            }
+
+            int lastError = Marshal.GetLastWin32Error();
+            if (lastError != Interop.Errors.ERROR_INVALID_PARAMETER)
```

Comment:
CoreCLR VM copy of this logic tries with `flags & 0xFFFFFF00` first and then tries again with `flags & 0xFF` if that failed with `ERROR_INVALID_PARAMETER`. 

This seems to be the handling the situation of missing KB2533623 on Win7. We do not support Win7 without KB2533623. I think we should:

- Delete the `ERROR_INVALID_PARAMETER` special handling here
- Delete the double call in CoreCLR VM (ie make CoreCLR VM match this)

Reply:
This is all very confusing.

It's not clear to me which one of the flags masks to keep (or none?). It's better if the interop team figures that out. I really don't want to spend time figuring out what KB2533623 was about and have another back-end-forth on this pre-existing code while ASP.NET team is waiting for this pull request to be merged.

I've now matched this to what CoreCLR does.

Reply:
Sounds good. We can keep this as is. It is probably not worth cleaning it up.